vendor/noise_handshake/src/handshake.rs

```diff
@@ -0,0 +1,209 @@
+use futures::*;
+use snow::Session;
+use tokio::io::{AsyncRead, AsyncWrite};
+
+const NOISE_MAX_SIZE: usize = 65535;
+
+#[allow(missing_debug_implementations)]
+pub enum Step {
+    Read {
+        enc_buffer: [u8; NOISE_MAX_SIZE],
+        len: usize,
+    },
+    Write {
+        buffer: [u8; NOISE_MAX_SIZE],
+        written_bytes: usize,
+        len: usize,
```

Comment:
🔧  let's give these more meaningful names:

-` buffer` change to `to_write`
- `len` change to `total_bytes`

In the Read part you could also change `len` to `read_bytes`